### ISSUE TITLE
Async Programming Model?

### ISSUE THREAD

**Issue by sherl0cks** [jupyter/kernel_gateway#270]
This a question.

tl;dr I'm looking for an asynchronous model for remotely invoking the execution of a cell in a notebook which is deployed via the gateway, 

I've run through the docs and examples. For short lived executions (e.g. < 10 seconds), I think the existing http personality works great out of the box. But I'm confused on how I should design for long running tasks e.g. > 10 seconds, where you would not want the client to block for an indefinite period if time. My reading of the websocket personality is that clients submit code to execute in the kernel, rather than invoking the execution of a cell in the notebook, like with the http personality.

**Comment by rolweber** [comment]
Is this related to #60?

**Comment by sherl0cks** [comment]
Not intentionally. They are in the same ballpark, though. If the websocket personality provides the feature describes in #60, that does also work for my needs. 

Can you confirm that #60 is still a feature request and not existing functionality?

**Comment by rolweber** [comment]
#60 is specifically for the notebook-http personality, where the executed code comes from a notebook configured on the server. This is still a possible extension and not existing functionality.

In websocket-http mode, we have websocket connections, but the code is sent by the client. That is not what you asked for above.

**Comment by sherl0cks** [comment]
Thanks for clarity @rolweber. I'm OK with this function being provided by #60 if that is the direction the project wants to take. I'd also be OK with something like returning a 202 Accepted by the endpoint along with a header for the client to poll the result. 

Thoughts?

**Comment by rolweber** [comment]
@sherl0cks: #60 is dormant, nobody's working on it. If you've got an idea how to implement what you need, go ahead and create a PR.

**Comment by rolweber** [comment]
I'm closing this in favor of #60.
